ibm/resource_ibm_event_streams_topic_test.go

```diff
@@ -0,0 +1,287 @@
+	retentionBytes := 10485760
+	retentionMs := 3600000
+	segmentBytes := 10485760
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMEventStreamsInstanceDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMEventStreamsTopicWithoutConfig(instanceName, serviceName, planID, location, topicName, partitions),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMEventStreamsTopicExists("ibm_event_streams_topic.es_topic", topicName),
+					resource.TestCheckResourceAttr("ibm_resource_instance.es_instance", "name", instanceName),
+					resource.TestCheckResourceAttr("ibm_resource_instance.es_instance", "service", serviceName),
+					resource.TestCheckResourceAttr("ibm_resource_instance.es_instance", "plan", planID),
+					resource.TestCheckResourceAttr("ibm_resource_instance.es_instance", "location", location),
+					resource.TestCheckResourceAttrSet("ibm_event_streams_topic.es_topic", "id"),
+					resource.TestCheckResourceAttr("ibm_event_streams_topic.es_topic", "name", topicName),
+					resource.TestCheckResourceAttr("ibm_event_streams_topic.es_topic", "partitions", strconv.Itoa(partitions)),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCheckIBMEventStreamsTopicWithConfig(instanceName, serviceName, planID, location, topicName, partitions, cleanupPolicy, retentionBytes, retentionMs, segmentBytes),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMEventStreamsTopicExists("ibm_event_streams_topic.es_topic", topicName),
+					resource.TestCheckResourceAttr("ibm_resource_instance.es_instance", "name", instanceName),
+					resource.TestCheckResourceAttr("ibm_resource_instance.es_instance", "service", serviceName),
+					resource.TestCheckResourceAttr("ibm_resource_instance.es_instance", "plan", planID),
+					resource.TestCheckResourceAttr("ibm_resource_instance.es_instance", "location", location),
+					resource.TestCheckResourceAttrSet("ibm_event_streams_topic.es_topic", "id"),
+					resource.TestCheckResourceAttr("ibm_event_streams_topic.es_topic", "name", topicName),
+					resource.TestCheckResourceAttr("ibm_event_streams_topic.es_topic", "partitions", strconv.Itoa(partitions)),
+					resource.TestCheckResourceAttr("ibm_event_streams_topic.es_topic", "config.cleanup.policy", cleanupPolicy),
+					resource.TestCheckResourceAttr("ibm_event_streams_topic.es_topic", "config.retention.bytes", strconv.Itoa(retentionBytes)),
+					resource.TestCheckResourceAttr("ibm_event_streams_topic.es_topic", "config.retention.ms", strconv.Itoa(retentionMs)),
+					resource.TestCheckResourceAttr("ibm_event_streams_topic.es_topic", "config.segment.bytes", strconv.Itoa(segmentBytes)),
+				),
+			},
+		},
+	})
+}
+
+var existingInstanceName = "hyperion-preprod-spp-a-service"
+
+func TestAccIBMEventStreamsTopicResourceWithExistingInstance(t *testing.T) {
```

Comment:
This test case may not be needed because we generally automate the testcase execution using jenkins....it requires an instance to exists

Reply:
I'd like to keep this case.
I think testing from your side: `TestAccIBMEventStreamsTopicResourceBasic` this creates both instance and topics.
while I will also setup regression test on our side to run the 4 cases, where existing instances/topics exist in our account.
```
make testacc TEST=./ibm TESTARGS='-run=TestAccIBMEventStreamsTopicDataSourceBasic'
make testacc TEST=./ibm TESTARGS='-run=TestAccIBMEventStreamsTopicResourceBasic'
make testacc TEST=./IBM TESTARGS='-run=TestAccIBMEventStreamsTopicResourceWithExistingInstance'
make testacc Tset=./IBM TESTARGS='-run=TestAccIBMEventStreamsTopicImport'
```